wire.go

```diff
@@ -0,0 +1,68 @@
+package opentracing
+
+// WireEncoding identifies a WireInjector/WireExtractor pair.
+type WireEncoding int
+
+const (
+	// WIRE_ENCODING_SPLIT_BINARY divides the encoded Span into two components:
+	//
+	//   1) The "tracer state" for the Span (for example, in Dapper this would
+	//      include a trace_id, a span_id, and a bitmask representing the
+	//      sampling status for the given trace)
+	//   2) Any Trace Attributes (per Span.SetTraceAttribute)
+	//
+	// The encoded data is separated in this way for a variety of reasons; the
+	// most important is to give OpenTracing users a portable way to opt out of
+	// Trace Attribute propagation entirely if they deem it a stability risk.
+	//
+	// The `carrier` for injection must be two pointers to `[]byte`s for the
```

Comment:
I really dislike how this RFC requires such subtle documentation of the `carrier` params (as I've called them here), esp since the compiler won't be able to check any of this for the user. Doubly bad since we're dealing with output parameters / pointers-to-slices.